src/sqlstore/champion.py

```diff
@@ -0,0 +1,120 @@
+from sqlalchemy import Column, Integer, String, Float, Text, CheckConstraint, PickleType, DateTime
+from sqlalchemy.sql import func
+from src.sqlstore.db import Base
+
+
+class SQLChampion(Base):
+    __tablename__ = "champion"
+
+    championId = Column(Integer, primary_key=True)
+    patchNumber = Column(Integer, primary_key=True)
+    championName = Column(String(100))
+    championTitle = Column(String(100))
+    infoAttack = Column(Integer)
+    infoDefense = Column(Integer)
+    infoMagic = Column(Integer)
+    infoDifficulty = Column(Integer)
+    tags = Column(PickleType)   # serialized list of tags (eg. [Marksman, Support] for Ashe)
+    partype = Column(String(150))   # type of mana or energy (eg. "Blood Well" for Aatrox)
+    patchWinRate = Column(Float, nullable=True)  # Represented as a percent
+    patchPlayRate = Column(Float, nullable=True)  # Represented as a percent
+    # TODO: should this be part of the primary key?
+    primaryRole = Column(String(50), nullable=True)  # Top, Mid...
+    # Maybe counters, abilities, Tier, maybe range, skill-shot-based, or not, cc-level.., trends in winrates,
+    # role flexibility, new skin released (higher playrate)
+    # -> this should not be saved in db, instead calculated server/analytics side imo
+    timeCreated = Column(DateTime(timezone=True), server_default=func.now())
+    lastUpdate = Column(DateTime(timezone=True), onupdate=func.now())
+
+    def init(self, championId: int, championName: str, championTitle: str, infoAttack: int, infoDefense: int,
+             infoMagic: int, infoDifficulty: int, tags, partype: str, patchWinRate: float = None,
+             patchPlayRate: float = None, patchNumber: int = None, role: str = None):
+        """
+
+        :param championId:
+        :param championName:
+        :param championTitle:
+        :param infoAttack:
+        :param infoDefense:
+        :param infoMagic:
+        :param infoDifficulty:
+        :param tags:
+        :param partype:
+        :param patchWinRate:
+        :param patchPlayRate:
+        :param patchNumber:
+        :param role:
+        :return:
+        """
+        self.championId = championId
+        self.championName = championName
+        self.championTitle = championTitle
+        self.infoAttack = infoAttack
+        self.infoDefense = infoDefense
+        self.infoMagic = infoMagic
+        self.infoDifficulty = infoDifficulty
+        self.tags = tags
+        self.partype = partype
+        self.patchWinRate = patchWinRate
+        self.patchPlayRate = patchPlayRate
+        self.patchNumber = patchNumber
+        self.primaryRole = role
+
+    def repr(self):
+        return f"<Champion {self.championName} ({self.key}) - {self.championTitle}>"
+
+
+class SQLChampionStats(Base):

```

Comment:
Adding foreign key to establish a relationship with the champion table (championId) and maybe patchNumber.
-> Integrity, Consistency

championId = Column(Integer, ForeignKey('champion.championId'), primary_key=True)
patchNumber = Column(Integer, ForeignKey('champion.patchNumber'), primary_key=True)

Reply:
and the primary key is just a SERIAL? a foreign key being a primary key as well is asking for problems